src/main/scala/viper/server/core/ViperCache.scala

```diff
@@ -280,22 +300,19 @@ object ViperCache extends Cache {
   def createCacheContent(
         backendName: String, file: String,
         p: Program, m: Method,
-        errors: List[AbstractVerificationError]): ViperCacheContent = {
-
-    implicit val key: String = getKey(backendName, file)
-    val loc_errs = errors.map(err =>
-      LocalizedError(err,
-        ViperCacheHelper.getAccessPath(err.offendingNode, p),
-        ViperCacheHelper.getAccessPath(err.reason.offendingNode, p),
-        backendName))
-    ViperCacheContent(loc_errs)
+        errors: List[AbstractVerificationError]): SerializedViperCacheContent = {
+
+    implicit val key: String = getKey(file = file, backendName = backendName)
+
+    implicit val formats: Formats = DefaultFormats.withHints(ViperCacheHelper.errorNodeHints(p, key))
+    SerializedViperCacheContent(write(ViperCacheContent(errors)))
```

Comment:
When do we serialize? I would have assumed that we only need to serialize the cache content when actually writing to disk.

Reply:
First, we serialize when writing the cache to file, to serialize the whole cache object. Second, we serialize these SerializedViperCacheContent when creating cache content, since we need to have access to the Program AST  to serialize and deserialize verification errors.

Reply:
Thanks. I missed the part that the AST is necessary